nelc/event-routing-backends
# feat: add aggregator events

**Description:**

This implements multiple transformers for events that I'm proposing in this [PR](https://github.com/open-craft/openedx-completion-aggregator/pull/173)

Progress events, those will be emitted  when the block has not reached the  100% of completion
-  `edx.completion_aggregator.progress.course`
- `edx.completion_aggregator.progress.chapter`
- `edx.completion_aggregator.progress.sequential`
- `edx.completion_aggregator.progress.vertical`

Completion events, those will be emitted when the block has reached the 100% of completion

-  `edx.completion_aggregator.completion.course`
- `edx.completion_aggregator.completion.chapter`
- `edx.completion_aggregator.completion.sequential`
- `edx.completion_aggregator.completion.vertical`

And basically I'm proposing this since the current "completion" implementation doesn't cover completion events  and that has the same limitation as the edx-completion plugin, that only works with unit components 


**JIRA:** Link to JIRA ticket

**Dependencies:** dependencies on other outstanding PRs, issues, etc.

**Merge deadline:** List merge deadline (if any)

**Installation instructions:** List any non-trivial installation
instructions.

**Testing instructions:**

1. Install this version of completion aggregator https://github.com/open-craft/openedx-completion-aggregator/pull/173
2. Raise the event by completing a section or subsection or unit 
3. Verify the the send task was executed 
4. Verify that your LRS is getting the events

![image](https://github.com/nelc/event-routing-backends/assets/36200299/84880370-df7e-476d-8d56-da5338d1e35f)

![image](https://github.com/nelc/event-routing-backends/assets/36200299/c03b25d2-afd2-4baa-a53d-0b5b6b19e6e1)

**Merge checklist:**

**Post merge:**
      finished.

**Author concerns:** List any concerns about this PR - inelegant
solutions, hacks, quick-and-dirty implementations, concerns about
migrations, etc.
